shell/src/chain_manager.rs

```diff
@@ -746,7 +759,34 @@ impl ChainManager {
                     // e.g. if we just start to bootstrap from the scratch, we dont want to spam other nodes (with higher level)
                     if self.is_bootstrapped {
                         match new_head_result {
-                            HeadResult::BranchSwitch => (/*"TODO: TE-174 sent current_branch message"*/),
+                            HeadResult::BranchSwitch => {
+                                let ChainManager {
+                                    peers,
+                                    chain_state,
+                                    identity_peer_id,
+                                    ..
+                                } = self;
+                                let header: &BlockHeader = &message.header().header;
+                                let chain_id = chain_state.get_chain_id();
+
+                                let block_hash: BlockHash = header.message_hash()?;
```

Comment:
@adonagy 
you dont need to compute hash `header.message_hash()?` (it is probably cached there, maybe we remove that cache in future),
you have it here already: `message.header().hash.clone()`

Reply:
Oh right! :)